anu_lms.install

```diff
@@ -66,3 +71,41 @@ function anu_lms_update_8001() {
     }
   }
 }
+
+/**
+ * Migrate is_highlight boolean field on image bullet list paragraphs to
+ * default color
+ */
+function anu_lms_update_8002() {
+  /** @var NodeInterface[] $lessons */
+  $lessons = Drupal::entityTypeManager()->getStorage('node')
+    ->loadByProperties(['type' => 'module_lesson']);
+  foreach ($lessons as $lesson) {
+    /** @var EntityReferenceRevisionsItem[] $sections */
+    $sections = $lesson->get('field_module_lesson_content')
+      ->referencedEntities();
+    foreach ($sections as $section) {
+      /** @var Paragraph $contents */
+      $contents = $section->get('field_lesson_section_content')
+        ->referencedEntities();
+      foreach ($contents as $content) {
+        if ($content->getType() == 'lesson_img_list') {
+          /** @var FieldItemList $highlight_field */
+          $highlight_field = $content->get('field_lesson_img_list_highlight');
+          if ($highlight_field->getString() == "1") {
+            $content->set('field_lesson_highlight_color', 'yellow');
+            $content->save();
+          }
+        }
+      }
+    }
+  }
+
+  /* @var $entityFieldManager Drupal\Core\Entity\EntityFieldManager */
+  $entityFieldManager = Drupal::service('entity_field.manager');
```

Comment:
Does this mean that we can remove lesson_img_list field from the configs? Otherwise the Features configs will keep offering users to import this config back, because it exists in the module's codebase. Or we can leave this field for now and delete all configs later - we'll be doing config cleanup anyway (i.e. delete old structure from ANU LMS v1 we used for content migration).

Reply:
shall i remove this then?

Reply:
I'd say yes, for now.